src/preamble.js

```diff
@@ -843,7 +843,13 @@ var splitModuleProxyHandler = {
       var imports = {'primary': Module['asm']};
       // Replace '.wasm' suffix with '.deferred.wasm'.
       var deferred = wasmBinaryFile.slice(0, -5) + '.deferred.wasm'
-      instantiateSync(deferred, imports);
+      if (Module["loadSplitModule"]) {
+        if (!Module["loadSplitModule"](prop, deferred, imports)) {
```

Comment:
Perhaps make the property argument the final one? 

That way, `instantiateSync` could be exact replacement for it.  We could even do something like this:

```
{{{ makeModuleReceiveWithVar('loadSplitModule', undefined, 'EXIT_RUNTIME' }}}
```

We could then simply call `loadSplitModule` here without any module checks since the above line would ensure that loadSplitModule exists.

Reply:
i changed this to 
```
 {{{ makeModuleReceiveWithVar('loadSplitModule', 'loadSplitModule', 'instantiateSync',  true) }}}
      loadSplitModule(deferred, imports, prop);
```

not sure if this is what you meant, @sbc100 would you please have a look?

Reply:
somehow this always replaces my code with instantiateSync

Reply:
ok i see that i have to add it to the INCOMING_MODULE_JS_API array